In [1]:
# Import required libraries 
import pandas as pd

In [2]:
# Read csv file 
dataset = pd.read_csv("./data/WELFake_Dataset.csv")

# Shows first five rows of dataset
dataset.head()

Unnamed: 0                                              title  \
0           0  LAW ENFORCEMENT ON HIGH ALERT Following Threat...   
1           1                                                NaN   
2           2  UNBELIEVABLE! OBAMA’S ATTORNEY GENERAL SAYS MO...   
3           3  Bobby Jindal, raised Hindu, uses story of Chri...   
4           4  SATAN 2: Russia unvelis an image of its terrif...   

                                                text  label  
0  No comment is expected from Barack Obama Membe...      1  
1     Did they post their votes for Hillary already?      1  
2   Now, most of the demonstrators gathered last ...      1  
3  A dozen politically active pastors came here f...      0  
4  The RS-28 Sarmat missile, dubbed Satan 2, will...      1

## Exploratory data analysis

In [3]:
# Let's check columns 
dataset.columns

Index(['Unnamed: 0', 'title', 'text', 'label'], dtype='object')

### 1. Inspect ratio of real and fake news in our dataset

In [4]:
# Inspecting ratio of real to fake news in our dataset
lab = dataset.label
print(f'Ratio of real and fake news:')
lab.value_counts(normalize=True).rename({ 0:'Fake', 1: 'Real'})

Ratio of real and fake news:


Real    0.514404
Fake    0.485596
Name: label, dtype: float64

### Observations:
#### 1. Real news Represents approximately 51.44% of the dataset while fake news represents 48.55% 
#### 2. With a little higher proportion of true news samples, the dataset seems to be fairly balanced between fake and real news.
#### 3. As we can see that the dataset isn't heavily skewed towards one class, the slight difference in proportions might still affect model performance, particularly if we are selecting a classifier which is sensitive to class imbalances.
#### 4. By providing a reasonably balanced representation of both fake and real news, models trained on this dataset may be able to avoid biases towards any particular class during training.

### 2. Drop extra unnamed coloumn

In [5]:
dataset.drop('Unnamed: 0', axis = 1, inplace = True)
dataset.head()

title  \
0  LAW ENFORCEMENT ON HIGH ALERT Following Threat...   
1                                                NaN   
2  UNBELIEVABLE! OBAMA’S ATTORNEY GENERAL SAYS MO...   
3  Bobby Jindal, raised Hindu, uses story of Chri...   
4  SATAN 2: Russia unvelis an image of its terrif...   

                                                text  label  
0  No comment is expected from Barack Obama Membe...      1  
1     Did they post their votes for Hillary already?      1  
2   Now, most of the demonstrators gathered last ...      1  
3  A dozen politically active pastors came here f...      0  
4  The RS-28 Sarmat missile, dubbed Satan 2, will...      1